controllers/modals/submission/ViewSubmissionMetadataHandler.inc.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * @file controllers/modals/submission/ViewSubmissionMetadataHandler.inc.php
+ *
+ * Copyright (c) 2014-2019 Simon Fraser University
+ * Copyright (c) 2003-2019 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ViewSubmissionMetadataHandler
+ * @ingroup controllers_modals_viewSubmissionMetadataHandler
+ *
+ * @brief Display submission metadata.
+ */
+
+// Import the base Handler.
+import('classes.handler.Handler');
+
+class ViewSubmissionMetadataHandler extends handler {
+
+	/**
+	 * Constructor.
+	 */
+	function __construct() {
+		parent::__construct();
+		$this->addRoleAssignment(array(ROLE_ID_REVIEWER), array('display'));
+	}
+
+	/**
+	 * @copydoc PKPHandler::authorize()
+	 */
+	function authorize($request, &$args, $roleAssignments) {
+		import('lib.pkp.classes.security.authorization.SubmissionAccessPolicy');
+		$this->addPolicy(new SubmissionAccessPolicy($request, $args, $roleAssignments));
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+
+	/**
+	 * Display metadata
+	 */
+	function display($args, $request) {
+		$submission = $this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION);
+		$reviewAssignment = $this->getAuthorizedContextObject(ASSOC_TYPE_REVIEW_ASSIGNMENT);
+		$context = $request->getContext();
+		$templateMgr = TemplateManager::getManager($request);
+		$publication = $submission->getCurrentPublication();
+
+		if ($reviewAssignment->getReviewMethod() != SUBMISSION_REVIEW_METHOD_DOUBLEBLIND) { /* SUBMISSION_REVIEW_METHOD_BLIND or _OPEN */
+			$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
+			$userGroups = $userGroupDao->getByContextId($context->getId());
+			$templateMgr->assign('authors', $publication->getAuthorString($userGroups));
+		}
+
+		$templateMgr->assign('publication', $publication);
+
+		if ($publication->getData('keywords')) {
+			$additionalMetadata[] = array(__('common.keywords'), implode(', ', $publication->getData('keywords')[AppLocale::getLocale()]));
```

Comment:
If no keywords are set, `$publication->getData('keywords')` will return `null`. It's also possible the locale will not be set. So I think you'll need to do something like:

```
$keywords = $publication->getData('keywords');
if (is_array($keywords) && isset($keywords[AppLocale::getLocale()])) {
  $additionalMetadata = ...
}
```

Reply:
Or you can do:

```
if ($publiciation->getLocalizedData('keywords') {
  $additionalMetadata == ...
}
```

That's probably cleaner.

Reply:
used the latter, check